src/globus_sdk/services/auth/client/confidential_client.py

```diff
@@ -294,3 +294,107 @@ def oauth2_token_introspect(
+            will be able to use. Should be false if the client is capable of keeping
+            secret credentials (such as clients running on a server) and true if it is
+            not (such as native apps). After creation this value is immutable.
+        :type public_client: bool
+        :param publicly_visible: If True, any authenticated entity can view it.
+            Otherwise, only entities in the same project as the client can view it.
+        :type publicly_visible: bool
+        :param redirect_uris: list of URIs that may be used in OAuth authorization
+            flows.
+        :type redirect_uris: iterable of str, optional
+        :param terms_and_conditions: URL of client's terms and conditions.
+        :type terms_and_conditions: str, optional
+        :param privacy_policy: URL of client's privacy policy.
+        :type privacy_policy: str, optional
+        :param required_idp: In order to use this client a user must have an identity
+            from this IdP in their identity set.
+        :type required_idp: str or uuid, optional
+        :param preselect_idp: This pre-selects the given IdP on the Globus Auth login
+            page if the user is not already authenticated.
+        :type preselect_idp: str or uuid, optional
+        :param create_params: Any additional parameters to be passed through.
+        :type create_params: dict, optional
+
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: pycon
+
+                    >>> ac = globus_sdk.AuthClient(...)
+                    >>> project_id = ...
+                    >>> r = ac.create_child_client(
+                    ...     "My client",
+                    ...     True,
+                    ...     True,
+                    ... )
+                    >>> client_id = r["client"]["id"]
+
+            .. tab-item:: Example Response Data
+
+                .. expandtestfixture:: auth.create_child_client
+
+            .. tab-item:: API Info
+
+                ``POST /v2/api/clients``
+
+                .. extdoclink:: Create Client
+                    :ref: auth/reference/#create_client
+        """
+        body: dict[str, t.Any] = {
+            "name": name,
+            "public_client": public_client,
+            "visibility": "public" if publicly_visible else "private",
+        }
+        if redirect_uris is not None:
+            body["redirect_uris"] = list(utils.safe_strseq_iter(redirect_uris))
+        if required_idp is not None:
+            body["required_idp"] = str(required_idp)
+        if preselect_idp is not None:
+            body["preselect_idp"] = str(preselect_idp)
+
+        # terms_and_conditions and privacy_policy must both be set or unset
+        if type(terms_and_conditions) is not type(privacy_policy):
+            raise exc.GlobusSDKUsageError(
+                "terms_and_conditions and privacy policy must both be set or unset"
+            )
```

Comment:
As noted elsewhere: this requirement isn't documented in the API docs. If it's accurate, we should target updating the Auth API docs to match the Auth behavior, too!